File: 3501-delete-nodes-from-linked-list-present-in-array/solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode modifiedList(int[] nums, ListNode head) {
        ListNode ans = new ListNode(0), temp = ans;
        ans.next = head;

        Set<Integer> n = new HashSet<>();
        
        for(int i = 0; i < nums.length; i++){
            n.add(nums[i]);
        }
        while(temp != null && temp.next != null){
            if(n.contains(temp.next.val)){
                temp.next = temp.next.next;
            }else temp = temp.next;
        }

        
       return ans.next;
    }
}

File: 3501-delete-nodes-from-linked-list-present-in-array/solution.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* modifiedList(vector<int>& nums, ListNode* head) {
        std::unordered_set<int> set(begin(nums), end(nums));

        ListNode* ptr = new ListNode{0};
        ptr->next = head;
        ListNode* ans = ptr;

        while(ptr != nullptr && ptr->next != nullptr){
            if(set.find(ptr->next->val) != set.end()){
                ptr->next = ptr->next->next;
            }
            else ptr = ptr->next;
        }
        return ans->next;
    }
};
